internal/transport/http_util.go

```diff
@@ -403,6 +432,24 @@ func newFramer(conn net.Conn, writeBufferSize, readBufferSize int, maxHeaderList
 	return f
 }
 
+func getWriteBufferPool(writeBufferSize int) *sync.Pool {
+	writeBufferMutex.Lock()
+	defer writeBufferMutex.Unlock()
+	size := writeBufferSize * 2
```

Comment:
What is the significant of the `* 2` here?

Reply:
IIRC this matches the legacy behavior, which always doubled for reasons nobody knows but we don't want to change now due to it affecting existing applications.

Reply:
That's fair. Do you mind expanding on the legacy behavior you are describing? It seems like if there is no shared buffer pool, the buffer is [allocated per the input size](https://github.com/grpc/grpc-go/blob/master/internal/transport/http_util.go#L314-L316). It only seems to be in the case of the buffer pool that we're allocating 2x the size (and this seems to be the PR the write buffer pool was added)?

Reply:
You may be right.. The old code multiplied by 2 when doing this allocation, so it seems we have changed the behavior for legacy users already:

https://github.com/grpc/grpc-go/blame/d524b409462c601ef3f05a7e1fba19755a337c77/internal/transport/http_util.go#L321

Reply:
Yeah - it just felt somewhat bizarre because this is the very PR that also removed that same ` * 2` in the default path (see line 321 above), but added the ` * 2` for the buffer pool path.

Reply:
If you think it's worth updating, I created https://github.com/grpc/grpc-go/pull/6983